app/controllers/species_list_controller.rb

```diff
@@ -390,6 +392,61 @@ def destroy_species_list # :norobots:
     end
   end
 
+  def add_remove_observations # :prefetch: :norobots:
+    pass_query_params
+    @id = params[:species_list].to_s
+    @query = find_query(:Observation)
+    if !@query
+      flash_error(:species_list_add_remove_no_query.t)
+      redirect_to(action: "list_species_lists")
+    end
+  end
+
+  def post_add_remove_observations # :prefetch: :norobots:
+    pass_query_params
+    id = params[:species_list].to_s
+    query = find_query(:Observation)
+    species_list = lookup_species_list(id)
+    flash_error(:species_list_add_remove_no_query.t) if !query
+    flash_error(:species_list_add_remove_bad_name.t(name: id.inspect)) if !species_list
+    if !query && !species_list
```

Comment:
Switching this to something like:

```
if species_list && query:
  do_add_remove_observations(species_list, query)
else:
  determine_redirect(species_list, query)
end
```

would be cleaner and clearer in my view.